apps/passport-client/components/modals/LegalTerms.tsx

```diff
@@ -0,0 +1,46 @@
+import { agreeTerms, LATEST_TERMS } from "@pcd/passport-interface";
+import { Spacer } from "@pcd/passport-ui";
+import { useCallback } from "react";
+import styled from "styled-components";
+import { appConfig } from "../../src/appConfig";
+import { useDispatch, useIdentity } from "../../src/appHooks";
+import { saveTermsAgreed } from "../../src/localstorage";
+import { Button, H2 } from "../core";
+import { TermsOfUse } from "../shared/TermsOfUse";
+
+export function LegalTermsModal() {
+  const dispatch = useDispatch();
+  const identity = useIdentity();
+
+  const onClick = useCallback(async () => {
+    const result = await agreeTerms(
+      appConfig.zupassServer,
+      LATEST_TERMS,
+      identity
+    );
+
+    if (result.success) {
+      dispatch({ type: "terms-agreed", version: result.value.version });
+    } else {
+      // Persist to local storage and sync this later
+      saveTermsAgreed(LATEST_TERMS);
+      dispatch({ type: "terms-agreed", version: result.value.version });
```

Comment:
this should have a loading spinner, and also handle server-side errors.

Reply:
Added a spinner around the remote API call.